### PR TITLE
Fix depwarn on 0.6

### PR DESCRIPTION
**Pull request by yuyichao**

### CODE REVIEW

**Comment by yuyichao** [comment]
AppVeyor script updated....

**Comment by yuyichao** [comment]
Hmmm, not sure what the CI failure is. Doesn't seem to have any error message?

**Comment by yuyichao** [comment]
...... The error should be fixed now.....

**Comment by yuyichao** [comment]
Hopefully really fixed this time.................

File: src/ZMQ.jl [hunk 1]
```diff
@@ -437,10 +437,13 @@ end
...
 # Convert message to string (copies data)
 unsafe_string(zmsg::Message) = Compat.unsafe_string(pointer(zmsg), length(zmsg))
-if VERSION < v"0.5-dev+4341"
-    bytestring(zmsg::Message) = unsafe_string(zmsg)
-else
-    @deprecate bytestring(zmsg::Message) unsafe_string(zmsg::Message)
+if isdefined(Base, :bytestring)
+    import Base: bytestring
+    if VERSION < v"0.5-dev+4341"
+        bytestring(zmsg::Message) = unsafe_string(zmsg)
```
Comment (stevengj) [inline src/ZMQ.jl]: On 0.5, seems like the analogous thing would be to define`String(zmsg::Message)`?

File: test/runtests.jl [hunk 2]
```diff
@@ -83,19 +83,25 @@ ZMQ.close(s1)
...
-   if VERSION < v"0.5-dev+4341"
-       @assert isempty(wait(reader))
-   else
-       @assert contains(wait(reader), "WARNING: bytestring(zmsg::Message) is deprecated")
-   end
-   eval(Base, :(have_color = $old_have_color)) # avoid control characters in output
+# deprecate bytestring(::Message), removed on 0.6
+isdefined(Base, :bytestring) && let olderr = STDERR
+    old_have_color = Base.have_color
+    eval(Base, :(have_color = false)) # avoid control characters in output
```
Comment (stevengj) [inline test/runtests.jl]: Seems like these should be in the `try/finally`.

File: src/ZMQ.jl (hunk 1 above)
Comment (yuyichao) [inline src/ZMQ.jl]: I think it's `unsafe_string`.

File: test/runtests.jl (hunk 2 above)
Comment (yuyichao) [inline test/runtests.jl]: I guess the setting shouldn't be in try but the restoring should be ideally in finally.... I'll do that when I'm back to a computer

**Pull request merged**